src/mj_sim.cpp

```diff
@@ -373,6 +379,90 @@ void MjSimImpl::setSimulationInitialState()
   mj_forward(model, data);
 }
 
+void MjSimImpl::makeDatastoreCalls()
+{
+  for(auto & r : robots)
+  {
+    // make_call for setting pd gains (for all joints)
+    controller->controller().datastore().make_call(
+        r.name + "::SetPDGains", [this, &r](const std::vector<double> & p_vec, const std::vector<double> & d_vec) {
+          const auto & rjo = controller->robots().robot(r.name).module().ref_joint_order();
+          if(p_vec.size() != rjo.size())
+          {
+            mc_rtc::log::warning("[mc_mujoco] {}::SetPDGains failed. p_vec size({})!=ref_joint_order size({})", r.name,
+                                 p_vec.size(), rjo.size());
+            return false;
+          }
+          if(d_vec.size() != rjo.size())
+          {
+            mc_rtc::log::warning("[mc_mujoco] {}::SetPDGains failed. d_vec size({})!=ref_joint_order size({})", r.name,
+                                 d_vec.size(), rjo.size());
+            return false;
+          }
+          r.kp = p_vec;
+          r.kd = d_vec;
+          return true;
+        });
+
+    // make_call for setting pd gains (by name)
+    controller->controller().datastore().make_call(
+        r.name + "::SetPDGainsByName", [this, &r](const std::string & jn, const double & p, const double & d) {
```

Comment:
just a quick note here, the datastore has an helper to call a registered function without specifying the argument types

As noted in the [doc](https://github.com/jrl-umi3218/mc_rtc/blob/master/include/mc_rtc/DataStore.h#L337) this treats arithmetic types specially so that they are passed by value.

Therefore I think you should register this with a signature of `(const std::string &, double, double)` so that it can be called easily via the helper:
```cpp
const std::string name = "my_joint_n";
double p = ...;
double d = ...;
datastore().call("MyRobot::SetPDGainsByName", name, p, d);
```

Reply:
Okay, I understand. I made the appropriate change.